### PR TITLE
Use build flag -mod=vendor instead of GO111MODULE=off

### LINKED ISSUE #280: Remove GO111MODULES environment variable from Makefile
I would leave setting this environment variable up to the user pre-build and just document it as a part of the install process. This also means you could set the minimum version of go to be `1.12` as well.

What do you think?

### LINKED ISSUE #281: Make sure to use the goflag -mod=vendor
Since we're using a vendor directory for building the product, we should make sure to use the `-mod=vendor` flag so that vendor dependencies are not downloaded to the user's `GOPATH`.

@mem did some of this work for `singularity` and perhaps could comment more here.

### PR DESCRIPTION
**Pull request by sashayakovtseva**
Closes #281 
Closes #280

### CODE REVIEW

**Comment by jscook2345** [comment]
@sashayakovtseva 

Just to verify, you'd like me to check that:

go1.11 and go1.12 built in `$GOPATH` works
go1.11 and go1.12 built in `$HOME` works

Is this correct?

**Comment by sashayakovtseva** [comment]
@jscook2345 Yes, that would be great. I checked 1.12, pretty sure 1.11 will work the same, so I downgraded Go version on `go.mod` file.

**Comment by jscook2345** [comment]
On `centos7`, `go1.11.10`, in `$GOPATH`:

```
[vagrant@localhost singularity-cri]$ pwd
/home/vagrant/go/src/github.com/sylabs/singularity-cri

[vagrant@localhost singularity-cri]$ make lint
golangci-lint run --disable-all \
--enable=gofmt \
--enable=goimports \
--enable=vet \
--enable=misspell \
--enable=maligned \
--enable=deadcode \
--enable=ineffassign \
--enable=golint \
--deadline=3m ./...

[vagrant@localhost singularity-cri]$ echo $?
0

[vagrant@localhost singularity-cri]$ GO111MODULE=on make
 GO bin/sycri
GOOS=linux go build -mod vendor -tags "selinux seccomp" -o ./bin/sycri ./cmd/server
build golang.org/x/crypto/openpgp/errors: cannot find module for path golang.org/x/crypto/openpgp/errors
make: *** [bin/sycri] Error 1
```

**Comment by jscook2345** [comment]
On `centos7`, `go1.11.10`, outside `$GOPATH`:

```
[vagrant@localhost singularity-cri]$ pwd
/home/vagrant/singularity-cri

[vagrant@localhost singularity-cri]$ make lint
golangci-lint run --disable-all \
--enable=gofmt \
--enable=goimports \
--enable=vet \
--enable=misspell \
--enable=maligned \
--enable=deadcode \
--enable=ineffassign \
--enable=golint \
--deadline=3m ./...
ERRO Running error: context loading failed: failed to load program with go/packages: go [list -e -json -compiled=true -test=true -export=false -deps=true -find=false -mod=vendor -- ./...]: exit status 1: build golang.org/x/crypto/openpgp/errors: cannot find module for path golang.org/x/crypto/openpgp/errors

make: *** [lint] Error 3

[vagrant@localhost singularity-cri]$ make
 GO bin/sycri
GOOS=linux go build -mod vendor -tags "selinux seccomp" -o ./bin/sycri ./cmd/server
build golang.org/x/crypto/openpgp/errors: cannot find module for path golang.org/x/crypto/openpgp/errors
make: *** [bin/sycri] Error 1
```

This required creating a config file so that `golangci-lint` would use the vendor directory. It also required a newer `golangci-lint` to use that option in the config (`1.17.1`)

```
[vagrant@localhost singularity-cri]$ cat .golangci.yml
run:
  modules-download-mode: vendor
```

**Comment by jscook2345** [comment]
So it looks like that one module is problematic for the build here. It appears in the `replace()` section of `go.mod`, but not in the `require` section. Is that the problem maybe?

**Comment by sashayakovtseva** [comment]
@jscook2345 Found this in modules doc:
```In Go 1.11, for direct dependencies a require directive is needed even when doing a replace. For example, if foo is a direct dependency, you cannot do replace foo => ../foo without a corresponding require for foo. If you are not sure what version to use in the require directive, you can often use v0.0.0 such as require foo v0.0.0. This was addressed in Go 1.12.
```

**Comment by jscook2345** [comment]
The `go1.11` builds worked for me! Great catch with that module fix.

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,15 +14,15 @@ jobs:
...
+            - singularity-cri
 
   gometalinter_check:
     <<: *defaults
     steps:
       - attach_workspace:
-          at: /go
+          at: ~/
       - run:
           name: Install golangci-lint v1.15
```
Comment (jscook2345) [inline .circleci/config.yml]: Still needs a config file for golanglint-ci that has the following content:

```
[vagrant@localhost singularity-cri]$ cat .golangci.yml
run:
  modules-download-mode: vendor
```

This also requires a newer golangci-lint. I tested with v1.17.1, the latest release.

**Comment by jscook2345** [comment]
The `go1.12` builds worked as well. Still had to set `GO111MODULE=on` in `$GOPATH`.

File: .circleci/config.yml (hunk 1 above)
Comment (sashayakovtseva) [reply .circleci/config.yml]: I don't understand how this will fix the error. The `go.mod` fix should fix linter as well...

Comment (sashayakovtseva) [reply .circleci/config.yml]: But updating linter is in general nice idea!

**Pull request merged**